### DISCUSSION TITLE
Running Airflow HA Scheduler w/ KubernetesExecutor in Multiple Kubernetes Clusters?

### DISCUSSION THREAD

**Discussion by JGoldman110** [apache/airflow#18455] [Q&A]
Is it possible to run 2 Airflow HA Schedulers, with KubernetesExecutor, where schedulers are running in separate Kubernetes Clusters?

**Comment by potiuk** [comment]
If you give the access to manage K8S resources between the clusters (so that for eample orphaned pods can be adopted) it might work - but this is a bit experimental and I am not sure it was ever tested, so not everything might work. But if you try and let us know if you have any problems we might be able to fix them,

**Reply by GaoJiaChengPaul** [reply]
Hi Potiuk,

Thanks for your quick reply.

We are trying to running 2 schedulers with different Kubernetes Clusters. However, the orphaned pods can not be adopted as the kubernetes job watcher will only monitor the pods in it own cluster. 

Let's say if **Scheduler A trigger a Task A in Cluster A**, and **Scheduler B is running in Cluster B**. And currently Task A is **running**.
Every time we try to deployment a new version of Airflow,  the **Scheduler A** will be shutdown and we will have a new **Scheduler C**.
And after the scheduler_health_check_threshold, **Scheduler A** will be marked as dead and Scheduler B will try to adopt Task A but its pod is in Cluster A not in Cluster B so actually Scheduler B will reset the Task A and finally the Task A will be marked as failed.

Btw, we have noticed that the reset logic in scheduler_job.py(https://github.com/apache/airflow/blob/main/airflow/jobs/scheduler_job.py)
seems that the actual **_resettable_states_** are not consistent with the upper comments. Do we need to change the  State.RUNNING to State.SCHEDULED?  Could you please help to advise? 
![image](https://user-images.githubusercontent.com/91484889/134943524-0b547509-0fe7-495f-bbd1-33fa9a9460ed.png)

**Reply by potiuk** [reply]
I am not that much of specialist there - so not sure if I can help with the state - but maybe others might chime in here - you can also ask the question about states in #kubernetes channel in Airflow Slack  - that's a simple question and you might get good answer there faster.. 

I think for supporting two Schedilers in two clusters you might open a Feature Request - sounds like a valid case and again - others who know more about it, maybe will be a able to comment on that.
